src/glassflow/pipelines.py

```diff
@@ -227,21 +246,37 @@ def consume_failed(self) -> operations.ConsumeFailedResponse:
 
         return res
 
-    def _get_headers(
-            self, request: dataclasses.dataclass,
-            req_content_type: Optional[str] = None
-    ) -> dict:
+    def _get_headers(self,
+                     request: dataclasses.dataclass,
+                     req_content_type: Optional[str] = None) -> dict:
 
         headers = utils.get_req_specific_headers(request)
         headers['Accept'] = 'application/json'
-        headers['Gf-Client'] = self.glassflow_client.glassflow_config.glassflow_client
-        headers['User-Agent'] = self.glassflow_client.glassflow_config.user_agent
+        headers[
+            'Gf-Client'] = self.glassflow_client.glassflow_config.glassflow_client
+        headers[
+            'User-Agent'] = self.glassflow_client.glassflow_config.user_agent
         headers['Gf-Python-Version'] = (f'{sys.version_info.major}.'
                                         f'{sys.version_info.minor}.'
                                         f'{sys.version_info.micro}')
 
-        if (req_content_type and req_content_type not in
-                ('multipart/form-data', 'multipart/mixed')):
+        if (req_content_type and req_content_type
+                not in ('multipart/form-data', 'multipart/mixed')):
             headers['content-type'] = req_content_type
 
         return headers
+
+    def _update_retry_delay(self, status_code: int):
+        if status_code == 200:
+            self._consume_retry_delay_current = self._consume_retry_delay_minimum
+        elif status_code == 204 or status_code == 429:
+            self._consume_retry_delay_current *= 2
+            self._consume_retry_delay_current = min(
+                self._consume_retry_delay_current,
+                self._consume_retry_delay_max)
+            self._consume_retry_delay_current += random.uniform(0, 0.1)
```

Comment:
There are 3 additional headers for 204 and 429 responses:
```
          headers:
            X-RateLimit-Limit:
              schema:
                type: integer
              description: Request limit per hour.
            X-RateLimit-Remaining:
              schema:
                type: integer
              description: The number of requests left for the time window.
            X-RateLimit-Reset:
              schema:
                type: string
                format: date-time
              description: The UTC date/time at which the current rate limit window resets.
```
This helps to create a precise delay.
Even more, not respecting these headers will penalize the delay to the next accepted request.

https://api.glassflow.dev/v1/docs#/default/consumeEvent